Core/MockInstantiationRewriter.cs

```diff
@@ -0,0 +1,104 @@
+//
+// THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY, FITNESS
+// FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER LIABILITY,
+// WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE SOFTWARE.
+//
+
+using System;
+using System.Linq;
+using Microsoft.CodeAnalysis;
+using Microsoft.CodeAnalysis.CSharp;
+using Microsoft.CodeAnalysis.CSharp.Syntax;
+
+namespace RhinoMocksToMoqRewriter.Core
+{
+  public class MockInstantiationRewriter : CSharpSyntaxRewriter
+  {
+    private readonly SemanticModel _model;
+
+    public MockInstantiationRewriter (SemanticModel model)
+    {
+      _model = model;
+    }
+
+    public override SyntaxNode? VisitInvocationExpression (InvocationExpressionSyntax node)
+    {
+      var compilationSymbol = _model.Compilation.GetTypeByMetadataName ("Rhino.Mocks.MockRepository");
+
+      if (compilationSymbol == null)
+      {
+        return node;
+      }
+
+      var symbol = _model.GetSymbolInfo (node).Symbol;
+      if (symbol == null)
+      {
+        return node;
+      }
+
+      var result = compilationSymbol
+          .GetMembers (symbol.Name)
+          .Where (
+              s => ParameterListEqualityComparer.Default
+                  .Equals ((s as IMethodSymbol)!.Parameters, (symbol.OriginalDefinition as IMethodSymbol)!.Parameters))
+          .FirstOrDefault (
+              s => ArgumentListEqualityComparer.Default
+                  .Equals ((s as IMethodSymbol)!.TypeArguments, (symbol.OriginalDefinition as IMethodSymbol)!.TypeArguments));
+
+      if (result == null)
+      {
+        return node;
+      }
+
+      var genericName = node.DescendantNodes().FirstOrDefault (n => n.IsKind (SyntaxKind.GenericName)) as GenericNameSyntax;
+      if (genericName == null || genericName.Identifier.Text != symbol.Name)
+      {
+        return node;
+      }
+
+      var typeArgumentList = SyntaxFactory.TypeArgumentList().AddArguments (genericName.TypeArgumentList.Arguments.First());
+
+      var argumentSyntaxList = SyntaxFactory.SeparatedList<ArgumentSyntax>();
+      if (symbol.Name == "GenerateStrictMock")
+      {
+        var strictArgument =
+            SyntaxFactory.Argument (
+                SyntaxFactory.MemberAccessExpression (
+                        SyntaxKind.SimpleMemberAccessExpression,
+                        SyntaxFactory.IdentifierName ("MockBehaviour"),
+                        SyntaxFactory.IdentifierName ("Strict"))
+                    .WithOperatorToken (
+                        SyntaxFactory.Token (SyntaxKind.DotToken)));
+
+        argumentSyntaxList = argumentSyntaxList
+            .Add (strictArgument);
+      }
+
+      if (symbol.Name == "GenerateStrictMock" && node.ArgumentList.Arguments.Count > 0)
```

Comment:
multiple usages of the same string literal should be merged into a string constant